server/rest/src/test/java/org/infinispan/rest/Http2Test.java

```diff
@@ -68,6 +69,28 @@ public void shouldUseHTTP2WithPriorKnowledge() throws Exception {
       clearTextUpgrade(true);
    }
 
+   @Test
+   public void shouldReportErrorCorrectly() throws Exception {
+      restServer = RestServerHelper.defaultRestServer()
+            .withKeyStore(KEY_STORE_PATH, "secret", "pkcs12")
+            .withTrustStore(KEY_STORE_PATH, "secret", "pkcs12")
+            .start(TestResourceTracker.getCurrentTestShortName());
+
+      RestClientConfigurationBuilder config = new RestClientConfigurationBuilder();
+
+      config.addServer().host(restServer.getHost()).port(restServer.getPort())
+            .protocol(HTTP_20).priorKnowledge(true)
+            .security().ssl().enable()
+            .trustStoreFileName(KEY_STORE_PATH).trustStorePassword("secret".toCharArray()).trustStoreType("pkcs12")
+            .keyStoreFileName(KEY_STORE_PATH).keyStorePassword("secret".toCharArray()).keyStoreType("pkcs12");
+
+      client = NettyHttpClient.forConfiguration(config.build());
+
+      FullHttpRequest getRequest = new DefaultFullHttpRequest(HTTP_1_1, GET, "invalid");
```

Comment:
shouldn't use HTTP2?

Reply:
The ```DefaultFullHttpRequest``` doesn't support it, but the client is configured as HTTP/2, so it will pass the request through a HTTP/2 pipeline